test/jdk/tools/jar/InputFilesTest.java

```diff
@@ -153,6 +153,51 @@ public void test6() throws IOException {
         jar("cf test.jar --release 9 -C test1 a -C test2 a");
     }
 
+    /**
+     * Containing non-existent file in the file list
+     * The final jar should not be created and correct error message should be caught.
+     * IOException is triggered as expected.
+     */
+    @Test(expectedExceptions = {IOException.class})
```

Comment:
Hello Weibing, I think using `expectedExceptions` here at the method level is too wide for this test, so might be better to remove it. I see that you use a catch block (at the right place) in this test method to verify you have received the correct `IOException`. What you could do is, don't rethrow it if it matches the expected IOException:

```java
        try {
            jar("cf test.jar existingTestFile.txt nonExistentTestFile.txt");
        } catch (IOException e) {
             // expected to have received an IOException due to tool
            // returning non-zero exit code
            Assert.assertEquals(e.getMessage().trim(), "nonExistentTestFile.txt : no such file or directory");
            Assert.assertTrue(Files.notExists(Path.of("test.jar")), "Jar file should not be created.");
        }
```